lib/mix/tasks/docs.ex

```diff
@@ -103,8 +103,8 @@ defmodule Mix.Tasks.Docs do
     * `:extras` - List of paths to additional Markdown (`.md` extension), Live Markdown
       (`.livemd` extension), Cheatsheets (`.cheatmd` extension) and plain text pages to
       add to the documentation. You can also specify keyword pairs to customize the
-      generated filename and title of each extra page; default: `[]`. Example:
-      `["README.md", "LICENSE", "CONTRIBUTING.md": [filename: "contributing", title: "Contributing"]]`
+      generated filename, title and source file of each extra page; default: `[]`. Example:
```

Comment:
Isn't the source file in this case the `key` here? So the example above would be written as:

```suggestion
  "CONTRIBUTING.md": [filename: "contributing", title: "Contributing", source: "CONTRIBUTING.mdx"]
```

Or are you telling me that you have .mdx files, which are converted to markdown, which are then converted to HTML?

If the second, maybe we should include custom preprocessors in ExDoc, so ExDoc is the one converting .mdx to .md, and you don't need a separate pipeline for it? I assume other projects could benefit from .mdx too?